packages/@lwc/engine/src/framework/api.ts

```diff
@@ -160,6 +161,7 @@ const ElementHook: Hooks = {
 const CustomElementHook: Hooks = {
     create: (vnode: VCustomElement) => {
         const { sel } = vnode;
+        attemptToRegisterTagName(sel);
```

Comment:
this could be optimize more in the future, instead of doing it here per every instance, we could do it somewhere else, maybe when the definition is created. the problem with doing this in `api.c()` is that it happens a lot more frequent than instance creation, because it runs on every rehydration process even if no changes has been detected, and no new instances are created.